src/views/ActivityTab.vue

```diff
@@ -0,0 +1,148 @@
+<!--
+  - @copyright Copyright (c) 2021 Louis Chemineau <louis@chmn.me>
+  -
+  - @author Louis Chemineau <louis@chmn.me>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div :class="{ 'icon-loading': loading }">
+		<!-- error message -->
+		<EmptyContent v-if="error" icon="icon-error">
+			{{ error }}
+		</EmptyContent>
+
+		<!-- activities content -->
+		<ul>
+			<Activity
+				v-for="activity in activities"
+				:key="activity.id"
+				:activity="activity" />
+		</ul>
+
+		<EmptyContent v-if="activities.length === 0 && !loading" icon="icon-activity">
+			{{ t('activity', 'No activity yet') }}
+		</EmptyContent>
+	</div>
+</template>
+
+<script>
+import { generateOcsUrl } from '@nextcloud/router'
+import axios from '@nextcloud/axios'
+import EmptyContent from '@nextcloud/vue/dist/Components/EmptyContent'
+
+import Activity from '../components/Activity'
+import ActivityModel from '../models/ActivityModel'
+
+import logger from '../logger'
+
+export default {
+	name: 'ActivityTab',
+	components: {
+		Activity,
+		EmptyContent,
+	},
+	data() {
+		return {
+			/**
+			 * @type {String}
+			 */
+			error: '',
+			/**
+			 * @type {Boolean}
+			 */
+			loading: true,
+			/**
+			 * @type {Object}
+			 */
+			fileInfo: null,
+			/**
+			 * @type {Array.<Object>}
+			 */
+			activities: [],
```

Comment:
We don't use types in data, I'm not convinced it really adds anything?
Most IDE already understand the default value in vue I think?